crates/graphql/src/schema/query_root.rs

```diff
@@ -441,6 +442,96 @@ impl QueryRoot {
         Ok(rows.pop().map(Into::into))
     }
 
+    #[graphql(description = "returns metadata_jsons matching the term")]
+    async fn metadata_json(
+        &self,
+        context: &AppContext,
+        #[graphql(description = "Search term")] term: String,
+        #[graphql(description = "Query limit")] limit: i32,
+        #[graphql(description = "Query offset")] offset: i32,
+    ) -> FieldResult<Vec<MetadataJson>> {
+        let search = &context.shared.search;
+
+        let result = search
+            .index("metadatas")
+            .search()
+            .with_query(&term)
+            .with_offset(offset.try_into()?)
+            .with_limit(limit.try_into()?)
+            .execute::<serde_json::value::Value>()
+            .await
+            .context("failed to load search result")?
+            .hits;
+
+        debug!("{:?}", result);
+        Ok(result
+            .into_iter()
+            .map(|r| {
+                let value = r.result;
+                MetadataJson {
```

Comment:
This is a good spot for a from trait.

Reply:
from trait for Serde::value to metadata json?

Reply:
You can deserialize a `Value` into a struct with `serde_json`